examples/ron/src/main.rs

```diff
@@ -0,0 +1,83 @@
+#![feature(proc_macro_hygiene)]
+
+#[macro_use] extern crate rocket;
+#[macro_use] extern crate serde_derive;
+
+#[cfg(test)] mod tests;
+
+use std::sync::Mutex;
+use std::collections::HashMap;
+
+use rocket::State;
+use rocket_contrib::ron::{Ron};
+
+// The type to represent the ID of a message.
+type ID = usize;
+
+// We're going to store all of the messages here. No need for a DB.
+type MessageMap = Mutex<HashMap<ID, String>>;
+
+#[derive(Serialize, Deserialize)]
+struct Message {
+    id: Option<ID>,
+    contents: String
+}
+
+#[derive(Serialize, Deserialize)]
+struct ErrorMessage {
```

Comment:
Is this added because it's harder to construct a `ron::Value` without shorthand like `json!`?

Let's rename this to `StatusMessage`, since it also encodes `ok` values. Additionally, we can change this to `status: &'static str`  and all of the `.to_owned()` calls will go away.

Reply:
That's correct, ron doesn't have a `ron!` equivalent yet. I still have my problems with the way rust handles strings, `.to_owned()` usually just works, but with `&'static str` it's a lot better.